app/code/Magento/Inventory/Model/Stock/Validator/SalesChannel/WebsiteCodeValidator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model\Stock\Validator\SalesChannel;
+
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventoryApi\Model\SalesChannelValidatorInterface;
+
+/**
+ * Check that website code is valid
+ */
+class WebsiteCodeValidator implements SalesChannelValidatorInterface
+{
+    /**
+     * @var ValidationResultFactory
+     */
+    private $validationResultFactory;
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * WebsiteCodeValidator constructor.
+     *
+     * @param ValidationResultFactory $validationResultFactory
+     * @param StoreManagerInterface   $storeManager
+     */
+    public function __construct(
+        ValidationResultFactory $validationResultFactory,
+        StoreManagerInterface $storeManager
+    ){
+        $this->validationResultFactory = $validationResultFactory;
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validate(SalesChannelInterface $salesChannel): ValidationResult
+    {
+        $errors[] = __('Website with code "%field" does not exist. Cannot add sales channel.'
+            , ['field' => $salesChannel->getCode()]);
+        $websites = $this->storeManager->getWebsites();
+        if (!empty($websites) && is_array($websites)) {
```

Comment:
I'm not sure that is a good idea to load ALL of the websites and iterate them in PHP
We can load website by code

Reply:
done.